packages/analytics/amplify_analytics_pinpoint/example/integration_test/enable_disable_test.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// A copy of the License is located at
+//
+//  http://aws.amazon.com/apache2.0
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+import 'dart:async';
+
+import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+import 'utils/mock_lifecycle_provider.dart';
+import 'utils/setup_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('enable/disable', () {
+    final mockLifecycleObserver = MockLifecycleProvider();
+
+    late final Stream<Map<String, Object?>> eventsStream;
+
+    setUpAll(() async {
+      eventsStream = await configureAnalytics(
+        appLifecycleProvider: mockLifecycleObserver,
+      );
+    });
+
+    tearDownAll(Amplify.reset);
+
+    test(
+      'disable prevents events from being auto flushed and sessions from being auto tracked',
+      () async {
+        await Amplify.Analytics.disable();
+
+        // Wait for all previous events to pass through
+        final streamSubscription = eventsStream.listen((event) {
+          expect(false, isTrue,
+              reason: 'Analytics disabled but events were still sent!');
+        });
+
+        const customEventName = 'my custom event type name';
+        final customEvent = AnalyticsEvent(customEventName);
+
+        await Amplify.Analytics.recordEvent(event: customEvent);
+
+        mockLifecycleObserver.triggerOnBackgroundListener();
+        mockLifecycleObserver.triggerOnForegroundListener();
+
+        streamSubscription.cancel();
+
+        await Amplify.Analytics.enable();
+        await Amplify.Analytics.flushEvents();
```

Comment:
What is the point of this if you're not testing anything afterwards? Shouldn't you test to see that events are now received?

Reply:
Thanks yeah there is no point now.  For context added that when all tests were together to avoid event from one polluting another but this is no longer necessary anymore.